app/assets/javascripts/service/datacontext.coffee

```diff
@@ -4,6 +4,15 @@
     $http.get( '/recipe.json' ).success ( data ) ->
       return data
 
+  @searchRecipes = (obj) ->
+    $http.get( '/recipe.json',{
```

Comment:
Just send obj, no need for creation of new object named params:
 $http.get( '/recipe.json',obj)